core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -875,23 +889,28 @@ object IO extends Serializable {
    */
   final def point[A](a: => A): IO[Nothing, A] = new Point(() => a)
 
+  /**
+   * Alias for point
+   */
+  final def succeed[A](a: => A): IO[Nothing, A] = point(a)
+
   /**
    * Creates an `IO` value that represents failure with the specified error.
    * The moral equivalent of `throw` for pure code.
    */
-  final def fail[E](error: E): IO[E, Nothing] = fail0(Cause.checked(error))
+  final def raiseChecked[E](error: E): IO[E, Nothing] = fail(Cause.checked(error))
```

Comment:
The only one I'm not really a fan of. It's easier to raise an untyped exception than a checked error.

Reply:
that's a good point -- I like `raiseChecked`, because it's a descriptive name
any suggestion how to rename `fail` so that it look more scary? `failFatally`? `failUnsafe`? `failHereBeDragons`?

Reply:
I don’t know to be honest 😣 Will think about it.

Reply:
after this change, `fail` doesn't take just any value, but a `Cause[E]`, which is good
if we want to make it even more uncomfortable to use (= less easy to misuse by people learning the library), we could rename it to `failWithACause`

Reply:
Very negative on this change, again, breaks nearly all written code yet I've basically come to associate ZIO with IO.fail...

Reply:
no opinion from my side -- just following the description of the ticket

Reply:
> any suggestion how to rename fail so that it look more scary? failFatally? failUnsafe? failHereBeDragons?

Should it be scary though? It's a nice well-defined operation with an all-around nice well-defined algebra to come with it. It can mean something bad, but it may also be used purely to short-circuit - and it's fine and completely safe. I think it's fine for it to be short. Quite unlike that nesty `terminate`!

Reply:
I also really like how concise it is to write `IO.fail` for checked errors. It's the one that people should use most of the time so it makes sense to have the shortest syntax for it.

Reply:
@ghostdogpr `fail` and `fail0` are nebulous terms and it's not obvious that one is for raising exceptions in a `checked` manner and the other for `unchecked` manner. and I also like that `raise` is part of the name, because that's what you do with exceptions -- you `raise` them, not `fail` them
newcomers to ZIO will have easier time with the new names